master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -624,12 +628,49 @@ private[celeborn] class Master(
     context.reply(OneWayMessageResponse)
   }
 
+  private def computeUserResourceConsumption(userIdentifier: UserIdentifier)
+      : ResourceConsumption = {
+    val current = System.currentTimeMillis()
+    if (userResourceConsumptions.containsKey(userIdentifier)) {
+      val resourceConsumptionAndUpdateTime = userResourceConsumptions.get(userIdentifier)
+      if (current - resourceConsumptionAndUpdateTime._2 > metricsResourceConsumptionInterval) {
+        val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
+          workerInfo.userResourceConsumption.asScala.get(userIdentifier)
+        }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+        userResourceConsumptions.put(userIdentifier, (newResourceConsumption, current))._1
```

Comment:
should return newResourceConsumption instead of previous one

Reply:
> should return newResourceConsumption instead of previous one

Done, I misunderstood the put's return...